File: yua-sdk-python/src/yua/resources/chat/messages.py
from __future__ import annotations

from typing import Any, Dict, List, Optional

from yua._core.api_client import APIClient


class Messages:
    def __init__(self, client: APIClient) -> None:
        self._client = client

    def list(self, thread_id: int) -> List[Dict[str, Any]]:
        return self._client.get(f"/api/chat/message?threadId={thread_id}")

    def create(
        self,
        *,
        thread_id: int,
        role: str = "user",
        content: str,
        attachments: Optional[List[Dict[str, Any]]] = None,
    ) -> Dict[str, Any]:
        body: Dict[str, Any] = {
            "threadId": thread_id,
            "role": role,
            "content": content,
        }
        if attachments:
            body["attachments"] = attachments
        return self._client.post("/api/chat/message", body)

File: yua-sdk/src/resources/chat/messages.ts
import type { APIClient } from "../../core/api-client";
import type { ChatMessage } from "../../types/chat";

export class Messages {
  private readonly client: APIClient;

  constructor(client: APIClient) {
    this.client = client;
  }

  async list(
    threadId: number,
    params?: { limit?: number },
  ): Promise<{ data: ChatMessage[] }> {
    const query = new URLSearchParams();
    query.set("threadId", String(threadId));
    if (params?.limit) query.set("limit", String(params.limit));

    const res = await this.client.get<{
      ok: boolean;
      messages: ChatMessage[];
    }>(`/api/chat/message?${query.toString()}`);

    return { data: res.messages ?? [] };
  }

  async create(
    threadId: number,
    params: {
      role: "user";
      content: string;
      attachments?: { file_id: string }[];
    },
  ): Promise<{ ok: boolean; messageId?: string }> {
    return this.client.post("/api/chat/message", {
      threadId,
      role: params.role,
      content: params.content,
      files: params.attachments?.map((a) => a.file_id),
    });
  }
}
